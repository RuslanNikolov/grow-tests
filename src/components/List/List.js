import React from "react";
import ListItem from "../ListItem/ListItem";

const List = props => {
  const { items } = props;

  return items.length > 0 ? (
    items.map(item => <ListItem key={item.id} {...item} />)
  ) : (
      <div className="no-items"> No items found.</div>
    );
};
export default List;
