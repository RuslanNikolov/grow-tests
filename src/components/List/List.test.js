import * as React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import List from './List'

const mockItems = [
    {
        id: 1,
        title: 'title1',
        url: 'url1',
        image: 'thumbnailUrl1'
    },
    {
        id: 2,
        title: 'title2',
        url: 'url2',
        image: 'thumbnailUrl2'
    },
]

describe('List', () => {
    let wrapper;
    it('should render the UI with items', () => {
        wrapper = shallow(
            <List
                items={mockItems}
            />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    it('should render the UI without items', () => {
        wrapper = shallow(
            <List
                items={[]}
            />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })


})