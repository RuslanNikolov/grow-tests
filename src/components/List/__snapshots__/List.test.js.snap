// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List should render the UI with items 1`] = `
Array [
  <ListItem
    id={1}
    image="thumbnailUrl1"
    key="1"
    title="title1"
    url="url1"
  />,
  <ListItem
    id={2}
    image="thumbnailUrl2"
    key="2"
    title="title2"
    url="url2"
  />,
]
`;

exports[`List should render the UI without items 1`] = `
<div>
   No items found.
</div>
`;
